docarray/array/mixins/find.py

```diff
@@ -144,7 +148,37 @@ def find(
                 index,
                 on=None,
             )
-        from docarray import Document, DocumentArray
+
+            if return_root:
+                from docarray.array.memory import DocumentArrayInMemory
+
+                def get_root_docs(da, docs):
+                    root_da_flat = da[...]
+                    da = DocumentArray()
+                    for doc in docs:
+                        result = doc
+                        while getattr(result, 'parent_id', None):
+                            result = root_da_flat[result.parent_id]
+                        da.append(result)
+                    return da
+
+                if not isinstance(self, DocumentArrayInMemory) and not all(
```

Comment:
Can we maybe split the special treatment of in-memory docarray into a separate function/method? No sure if it can be done easily, but if it can, I would do it